front/pages/w/[wId]/assistant/[cId]/index.tsx

```diff
@@ -91,7 +93,24 @@ export default function AssistantConversation({
         mainTag.scrollTo(0, 0);
       }
     }
-  }, [router.query, setConversationKey, initialConversationId]);
+
+    const handleRouteChange = () => {
```

Comment:
Why not doing this `setAgentIdToMention` in the `useEffect` from above which does trigger when `router.query` changes?